### PR TITLE
Adding mmd connections with auto retry on disconnect

### PR DESCRIPTION
**Pull request by adithyant**

### CODE REVIEW

File: mmd/mmd.go [hunk 1]
```diff
@@ -256,6 +313,14 @@ func (c *Conn) Close() {
 	close(c.writeChan)
 }
 
+func onDisconnect(c *Conn) {
```
[the remark this answers is not in the document]
Comment (adithyant) [reply mmd/mmd.go]: Makes sense for it to be a part of the conn. Made the changes.

**Comment by adithyant** [comment]
@dbudworth Made the code review changes. I had one question. Does it make sense to panic when we are not able to resolve the address rather than retrying again?

File: mmd/mmd.go [hunk 2]
```diff
@@ -331,33 +334,34 @@ func cleanupReader(c *Conn) {
...
-		go writer(c)
-		go reader(c)
-		handshake := []byte{1, 1}
-		handshake = append(handshake, c.config.AppName...)
+func (c *Conn) reconnect() {
+	err := c.resetSocket()
 
-		c.WriteFrame(handshake)
+	if err != nil {
+		log.Panicln("Failed to reconnect: ", err)
```
Comment (dbudworth) [inline mmd/mmd.go]: The return here is not necessary as log.Panicln will not return

**Comment by adithyant** [comment]
@dbudworth I tried to move all the repeated stuff into a common place. Let me know if this looks good.

**Pull request merged**